pkg/dogstatsd/parse.go

```diff
@@ -139,20 +143,32 @@ func (p *parser) parseMetricSample(message []byte) (dogstatsdMetricSample, error
 		}
 	}
 
+	// now, look for extra fields supported by dogstatsd
+	// sample rate, tags, container ID, timestamp, ...
+
 	sampleRate := 1.0
 	var tags []string
 	var containerID []byte
 	var optionalField []byte
+	var timestamp time.Time
 	for message != nil {
 		optionalField, message = nextField(message)
 		switch {
+		// tags
 		case bytes.HasPrefix(optionalField, tagsFieldPrefix):
 			tags = p.parseTags(optionalField[1:])
+		// sample rate
 		case bytes.HasPrefix(optionalField, sampleRateFieldPrefix):
 			sampleRate, err = parseMetricSampleSampleRate(optionalField[1:])
 			if err != nil {
 				return dogstatsdMetricSample{}, fmt.Errorf("could not parse dogstatsd sample rate %q", optionalField)
 			}
+		// timestamp
+		case bytes.HasPrefix(optionalField, timestampFieldPrefix):
+			if ts, err := strconv.ParseInt(string(optionalField[len(timestampFieldPrefix):]), 10, 0); err == nil {
+				timestamp = time.Unix(ts, 0)
+			}
```

Comment:
When `err != nil` why not returning an error like for an invalid sample rate?

Reply:
True. I'll change this.